dlt/sources/filesystem/readers.py

```diff
@@ -0,0 +1,129 @@
+from typing import TYPE_CHECKING, Any, Dict, Iterator, Optional
+
+from dlt.common import json
+from dlt.common.typing import copy_sig_any
+from dlt.sources import TDataItems, DltResource, DltSource
+from dlt.sources.filesystem import FileItemDict
+
+from .helpers import fetch_arrow, fetch_json
+
+
+def _read_csv(
+    items: Iterator[FileItemDict], chunksize: int = 10000, **pandas_kwargs: Any
+) -> Iterator[TDataItems]:
+    """Reads csv file with Pandas chunk by chunk.
+
+    Args:
+        chunksize (int): Number of records to read in one chunk
+        **pandas_kwargs: Additional keyword arguments passed to Pandas.read_csv
+    Returns:
+        TDataItem: The file content
+    """
+    import pandas as pd
+
+    # apply defaults to pandas kwargs
+    kwargs = {**{"header": "infer", "chunksize": chunksize}, **pandas_kwargs}
+
+    for file_obj in items:
+        # Here we use pandas chunksize to read the file in chunks and avoid loading the whole file
+        # in memory.
+        with file_obj.open() as file:
+            for df in pd.read_csv(file, **kwargs):
+                yield df.to_dict(orient="records")
+
+
+def _read_jsonl(items: Iterator[FileItemDict], chunksize: int = 1000) -> Iterator[TDataItems]:
```

Comment:
@rudolfix 
I'd suggest adding another reader for `.json` files. It's quite easy and it's being requested from time to time.